### ISSUE TITLE
release needed for missing url bug

### ISSUE THREAD

**Issue by primaeval** [vonH/plugin.video.iplayerwww#106]
The missing url bug is still there. Could you get a release out.

http://i.imgur.com/mUJUfNb.png

**Comment by CaptainTK** [comment]
I know this is still open. From my point of view this is a minor bug, which does not require a quick fix.

Also, I am out of town this week and a release would not be straightforward. Since I merged your Red Button changes to master, I would first need to roll back to the previous state. I don't want to release with Red Button because it didn't work for me when I tested (maybe simply because there was nothing on).

Long story short: No release this week unless something significant breaks.

**Comment by primaeval** [comment]
It looks really ugly to me. Like a hairy wart. ;) It is the only thing wrong with the addon at the moment and it really stands out as buggy.

You have to look at the schedule for the Red Button on the Sport page link to find when something is on. Do a find for "Red Button" on here: http://www.bbc.co.uk/sport/17151954
Others seem to be happy with the Red Button addition as it works for them too.

Enjoy your trip.

**Comment by primaeval** [comment]
Just tuck the Red Button away behind an optional enable button in Settings  if you think it will confuse people.

**Comment by CaptainTK** [comment]
I have a rather high tolerance against ugly things. It comes for free with looking in the mirror every morning. ;-)

Red Button would be a great addition to the add-on. I just want to make sure that it works and that I understand how it works. Hiding it is not a good idea. I believe most users never make it to the preferences menu.

**Comment by CaptainTK** [comment]
OK, things are starting to look better. I finally had the time to understand some of the issues I had and found that they are not related to your code, but to my setup. Thus, Red Button looks good to go for me.

I just added a warning dialog in my latest commit, can you please take a look and let me know what you think?

**Comment by primaeval** [comment]
The dialog is a good idea. It needs a setting backing it up so it doesn't appear again, like the EU cookie law. If anyone is using it from an epg and the dialog pops up all the time it would be a bit of a pain.

**Comment by primaeval** [comment]
edit: It doesn't come up from a favourite shortcut so it might be ok like it is.

**Comment by CaptainTK** [comment]
I'd love to add a "don't show this again" checkbox, but it seems the yesno-dialog in xbmcgui does not support that.

The only other option that occurs to me would be an xbmcgui.notification dialog. This does not require user interaction, but will always pop-up for a certain amount of time. Seems even more annoying to me.

**Comment by primaeval** [comment]
A checkbox in the settings would be enough to show they have read the warning.

**Comment by CaptainTK** [comment]
But then they need to go to settings to switch it off. And this information should be in the dialog itself, which makes it bloated.

That's pretty annoying.

**Comment by primaeval** [comment]
That's Kodi for you. I've been making an imdb search addon recently and I keep coming across these sort of gui limitations. I just tell myself either just accept it or rewrite a complete skin and c++ module from scratch. ;)

**Comment by CaptainTK** [comment]
:)

All right, I'll leave it as it is now and see if there are a lot of complaints. In that case, we can always add an option in settings to disable the dialog.
